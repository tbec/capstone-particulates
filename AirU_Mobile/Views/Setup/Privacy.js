/**
 * Screen for privacy settings
 **/ 

import React, {Component} from 'react';
import {Text, View, TouchableHighlight,Image} from 'react-native';
import NavBar from '../../Components/NavBar'
import styles from '../../StyleSheets/Styles'

export default class Privacy extends Component<Props> {
    constructor(props) {
        super(props);
        this.state = {privacy: null}
    }

    // after selecting option, sets as appropriate 
    setPrivacy(value) {
        this.setState({privacy: value})
        this.props.navigation.navigate('Confirmation');
    }

    render() {
        return (
            <View style={styles.mainView}>
                <View style={{flex: 5, alignContent: 'center', justifyContent: 'center'}}>
                    <Text style={{flex: 2}}>Your sensor has been successfully connected!
                        Would you like to make your sensor visable publically? 
                        This will allow people to view your sensor's location on a map and for researchers to directly use and access 
                        the data your sensor gathers.
                    </Text>
                    <Text/>
                    <Image source={require('../../Resources/SensorPrivacy.png')} style={{width: '100%', height: '100%', flex: 4, alignContent: 'center'}}/>
                </View>
                {/* choice section */}
                <View style={[styles.home, {flex: 3}]}>
                    <TouchableHighlight style={styles.button} onPress={() => this.setPrivacy(true)}>
                        <Text>Allow</Text>
                    </TouchableHighlight>
                    <TouchableHighlight style={styles.button} onPress={() => this.setPrivacy(false)}>
                        <Text>Deny</Text>
                    </TouchableHighlight>
                </View>
                <NavBar navigation={this.props.navigation} previous='ConnectionSetup'/>
            </View>
        );
    }
}