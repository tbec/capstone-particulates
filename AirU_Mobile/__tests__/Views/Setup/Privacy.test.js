import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Privacy from '../../../Views/Setup/Privacy'

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

  jest.mock('TouchableHighlight', () => {
    const mockComponent = require('react-native/jest/mockComponent');
    return mockComponent('TouchableHighlight');
  });
  
/* Tests */
test('Snapshot ConnectionSetup Setup', () => {
    const snap = renderer.create(<Privacy/>).toJSON();
    expect(snap).toMatchSnapshot();
})
