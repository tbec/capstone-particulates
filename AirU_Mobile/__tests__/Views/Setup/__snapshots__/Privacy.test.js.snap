// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ConnectionSetup Setup 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "height": "80%",
      "justifyContent": "space-between",
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 2,
        }
      }
    >
      Your sensor has been successfully connected! Would you like to make your sensor visable publically? This will allow people to view your sensor's location on a map and for researchers to directly use and access the data your sensor gathers.
    </Text>
    <Text />
    <Image
      source={
        Object {
          "testUri": "../../../Resources/SensorPrivacy.png",
        }
      }
      style={
        Object {
          "alignContent": "center",
          "flex": 4,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 100,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        },
        Object {
          "flex": 3,
        },
      ]
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "space-evenly",
          "padding": 1,
          "width": 150,
        }
      }
      testID="setPrivacyTrue"
      underlayColor="black"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Allow
      </Text>
    </TouchableHighlight>
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "space-evenly",
          "padding": 1,
          "width": 150,
        }
      }
      testID="setPrivacyFalse"
      underlayColor="black"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Deny
      </Text>
    </TouchableHighlight>
  </View>
  <View
    style={
      Object {
        "alignContent": "space-between",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={30}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "transparent",
          "borderColor": "black",
          "flex": 1,
          "justifyContent": "flex-start",
          "padding": 10,
        }
      }
      underlayColor="yellow"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 40,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï„’
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
